plugins/dell-dock2/fu-dell-dock2-wtpd.c

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2024 Dell Inc.
+ * All rights reserved.
+ *
+ * This software and associated documentation (if any) is furnished
+ * under a license and may only be used or copied in accordance
+ * with the terms of the license.
+ *
+ * This file is provided under a dual MIT/LGPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ * Dell Chooses the MIT license part of Dual MIT/LGPLv2 license agreement.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-dell-dock2-common.h"
+
+struct _FuDellDock2Wtpd {
+	FuDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuDellDock2Wtpd, fu_dell_dock2_wtpd, FU_TYPE_DEVICE)
+
+static gchar *
+fu_dell_dock2_wtpd_ver_string(guint32 wtpd_version)
+{
+	/* guint32 BCD */
+	return g_strdup_printf("%02x.%02x.%02x.%02x",
+			       (wtpd_version >> 0) & 0xff,
+			       (wtpd_version >> 8) & 0xff,
+			       (wtpd_version >> 16) & 0xff,
+			       (wtpd_version >> 24) & 0xff);
+}
+
+static gboolean
+fu_dell_dock2_wtpd_setup(FuDevice *device, GError **error)
+{
+	FuDevice *proxy = fu_device_get_proxy(device);
+	g_autofree gchar *dynamic_version = NULL;
+	g_autofree gchar *instance_id = NULL;
+	g_autofree const gchar *devname = NULL;
+	g_autofree const gchar *guid = NULL;
+	guint32 wtpd_version;
+	guint8 devtype = DELL_DOCK2_EC_DEV_TYPE_WTPD;
+	DellDock2BaseType dock_type = fu_dell_dock2_ec_get_dock_type(proxy);
+	guint8 dock_sku = fu_dell_dock2_ec_get_dock_sku(proxy);
+
+	/* name */
+	devname = g_strdup_printf("%s", fu_dell_dock2_ec_devicetype_to_str(devtype, 0, 0));
+	fu_device_set_name(device, devname);
+
+	/* IDs */
+	instance_id = g_strdup_printf("WTPD\\DOCKTYPE_%02x&DOCKSKU_%02x", dock_type, dock_sku);
```

Comment:
this needs to be built, not hardcoded. e.g. fu_device_build_instance_id

Reply:
Fixed.